### ISSUE TITLE
Qtile Crash - Multiple Errors

### ISSUE THREAD

**Issue by JonBoyleCoding** [qtile/qtile#3098]
I just had qtile crash while idling, and found these errors in the log file. I believe I'm on the latest version as of posting, the qtile version is reported as `0.18.2.dev132+g18af6f31`.

This seems to be the cause of the actual crash - these lines are quite a number of times over the course of 10 seconds or so.

```
2021-12-15 20:46:36,553 ERROR libqtile base.py:on_done():L632 Failed to reschedule.
Traceback (most recent call last):
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/lib/python3.9/site-packages/libqtile/core/manager.py", line 218, in async_loop
    self.load_config(initial=True)
  File "/usr/lib/python3.9/site-packages/libqtile/core/manager.py", line 163, in load_config
    os.remove(self._state)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/qtile-state'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/libqtile/widget/base.py", line 624, in on_done
    self.update(result)
  File "/usr/lib/python3.9/site-packages/libqtile/widget/base.py", line 535, in update
    old_width = self.layout.width
  File "/usr/lib/python3.9/site-packages/libqtile/drawer.py", line 80, in width
    return self.layout.get_pixel_size()[0]
  File "/usr/lib/python3.9/site-packages/libqtile/pangocffi.py", line 135, in get_pixel_size
    pango.pango_layout_get_pixel_size(self._pointer, width, height)
TypeError: initializer for ctype 'PangoLayout *' must be a cdata pointer, not NoneType
2021-12-15 20:46:36,562 ERROR libqtile start.py:start():L91 Qtile crashed
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/libqtile/scripts/start.py", line 89, in start
    q.loop()
  File "/usr/lib/python3.9/site-packages/libqtile/core/manager.py", line 195, in loop
    asyncio.run(self.async_loop())
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/lib/python3.9/site-packages/libqtile/core/manager.py", line 218, in async_loop
    self.load_config(initial=True)
  File "/usr/lib/python3.9/site-packages/libqtile/core/manager.py", line 163, in load_config
    os.remove(self._state)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/qtile-state'
```

I don't know if this is related, but I also noted this error in the same time period at the beginning of the errors, but this appears a number of times in the log where no crash happens.

```
2021-12-15 20:46:34,464 ERROR libqtile base.py:on_done():L632 Failed to reschedule.
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/libqtile/widget/base.py", line 624, in on_done
    self.update(result)
  File "/usr/lib/python3.9/site-packages/libqtile/widget/base.py", line 541, in update
    self.draw()
  File "/usr/lib/python3.9/site-packages/libqtile/widget/base.py", line 507, in draw
    self.drawer.draw(offsetx=self.offsetx, offsety=self.offsety, width=self.width)
  File "/usr/lib/python3.9/site-packages/libqtile/backend/base.py", line 708, in draw
    self._draw(offsetx, offsety, width, height)
  File "/usr/lib/python3.9/site-packages/libqtile/backend/x11/drawer.py", line 157, in _draw
    self.qtile.core.conn.conn.core.CopyArea(  # type: ignore
  File "/usr/lib/python3.9/site-packages/xcffib/xproto.py", line 3035, in CopyArea
    return self.send_request(62, buf, is_checked=is_checked)
  File "/usr/lib/python3.9/site-packages/xcffib/__init__.py", line 391, in send_request
    seq = self.conn.send_request(flags, xcb_parts + 2, xcb_req)
  File "/usr/lib/python3.9/site-packages/xcffib/__init__.py", line 569, in wrapper
    self.invalid()
  File "/usr/lib/python3.9/site-packages/xcffib/__init__.py", line 556, in invalid
    raise XcffibException("Invalid connection.")
xcffib.XcffibException: Invalid connection.
```

Any thoughts on what caused the issue? Thanks.

**Comment by m-col** [comment]
How are you starting Qtile? That first error suggests you are trying to start it with some pre-existing state. The other error, I *think*, appears when restarting or config reloading and we aren't clearing something up completely. It likely won't actually cause any problems and so will only be detectable in the logs. Do you restart/reload occasionally?

**Comment by JonBoyleCoding** [comment]
I'm starting qtile via lightdm, I believe the desktop file has been placed by Arch package, but it's just running "qtile start". I don't know how the system was fully in that state, but I came to the machine after leaving it for a while and it had just crashed and gone to the logon screen. 

I haven't had the crash again, although I had to revert back to the most recently released version as for some reason it wasn't letting me login on the most recent git version on this machine (my config is working fine on another machine though so I don't know why that is).

As for the second one, I don't really restart at all unless I'm hacking my config, which I wasn't then. I seem to get quite a number of xcb errors though, perhaps for completely different reasons.


As this seems to have been a one-off issue, unless you have any further thoughts, I shall close this down. Thanks for taking the time to look at this.
